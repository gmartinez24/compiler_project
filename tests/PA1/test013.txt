// Spencer Banasik
// Ensure that subtraction vs negation is valid, that our identifiers follow
// the constraints set by the grammar, if tabs are included in whitespace,
// a tricky comment scanning case,
// and finally, if we handle how windows adds newlines for portability reasons
// the character used is \r which always appears before a \n.

main

int ourfirst_var = - 15.5;
float 5is_this_valid = -15.5; // The name shouldn't be valid, a 5 and the identifier is the result
int _how_about_this = 5.; // The name shouldn't be valid, an error and the identifier is the result

Start				tabs

// Can the comments handle /*/ or does it accept that as a start and end
/*/* The below is just an experiment! */*/

// does a backslash escape the newline? \
ifnot,thisshouldbe45,three_identifiers


