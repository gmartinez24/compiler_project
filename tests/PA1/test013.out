MAIN
INT
IDENT	ourfirst_var
ASSIGN
SUB
FLOAT_VAL	15.5
SEMICOLON
FLOAT
INT_VAL	5
IDENT	is_this_valid
ASSIGN
FLOAT_VAL	-15.5
SEMICOLON
INT
ERROR
IDENT	how_about_this
ASSIGN
ERROR
SEMICOLON
IDENT	Start
IDENT	tabs
MUL
DIV
IDENT	ifnot
COMMA
IDENT	thisshouldbe45
COMMA
IDENT	three_identifiers
EOF